src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.cpp

```diff
@@ -723,6 +729,15 @@ std::shared_ptr<Repeated> Snapshot::tryMergeTriangles(const std::vector<Group::G
         return {};
     }
 
+    if (prods.size() < m_ctx.keep_blocks) {
```

Comment:
Can add a property to enable those 2 checks so it doesn't break the default config

Reply:
@dmatveev thoughts?

Reply:
I've convinced myself that this check is harmless